spec/java/util/Optional.lsl

```diff
@@ -71,27 +110,27 @@ import "java/util/stream/_interfaces.lsl";
 
     // constructors
 
-    @private constructor Optional ()
+    @private constructor Optional (@target @Parameterized(["T"]) self: Optional)
     {
         action ERROR("Private constructor call");
-        /*assigns self.value;
-        ensures self.value == null;
+        /*assigns this.value;
+        ensures this.value == null;
 
-        value = null;*/
+        this.value = null;*/
     }
 
 
-    @private constructor Optional (obj: T)
+    @private constructor Optional (@target @Parameterized(["T"]) self: Optional, obj: T)
     {
         action ERROR("Private constructor call");
-        /*required obj != null;
-        assigns self.value;
-        ensures self.value == obj;
+        /*requires obj != null;
+        assigns this.value;
+        ensures this.value == obj;
 
         if (obj == null)
-            self._throwNPE();
+            {_throwNPE();}
```

Comment:
this._throwNPE() ?

Reply:
Or, this is @static; I found this below

Reply:
Yes, this subroutine will be generated as `static` method and do not require `this`.
The exact reason is much simpler: current version of the parser analyzes this expression incorrectly: `this._throwNPE` will be the resolved name of the subroutine which is wrong and does not exist. Thus, plain `_throwNPE()` is the only option for now.